sonar-scala-plugin/src/test/resources/checks/StringLiteralDuplicatedScala.scala

```diff
@@ -0,0 +1,61 @@
+class MyClass {
+  def foo(str: String) = {
+    str match {
+      case "Alice" =>
+        val message = "Hello dear" + "Alice" // Noncompliant
+        message
+      case "Bob" =>
+        val message = "Hello dear" + "Bob"
+        message
+      case _ =>
+        val message = "Hello dear" + "Guest"
+        message
+    }
+  }
+
+  def bar() : Unit = {
+    val greeting1 = "Welcome to our site!" // Compliant, literal only appears twice
+    val greeting2 = "Welcome to our site!"
+  }
+
+  def baz() : Unit = {
+    val farewell1 = "Goodbye" // Compliant, literal is a single word
+    val farewell2 = "Goodbye"
+    val farewell3 = "Goodbye"
+  }
+
+
+  def foo2 (): Unit = {
+    val a = "string_literal5" // Compliant
+    val b = "string_literal5"
+    val c = "string_literal5"
+  }
+
+  def qux(): Unit = {
+    val x = "as d" // Compliant, literal length smaller than threshold
+    val y = "as d"
+    val z = "as d"
+  }
+
+
+  object Compliant {
+    @SuppressWarnings("scala:S1192") // Compliant
+    def a(): String =  "Hello2!" + "Hello2!" + "Hello2!" // Noncompliant
+    @SuppressWarnings("scala:S1192")
+    def b(): String =  "Welcome2!" + "Welcome2!" + "Welcome2!" // Noncompliant
+    @SuppressWarnings("scala:S1192")
+    def c(): String = "ByeBye!" +  "ByeBye!" + "ByeBye!" // Noncompliant
+  }
+  object Compliant2 {
+    @SuppressWarnings("scala:S1192") // Compliant
+    def a(): String = "scala:S1192" + "scala:S1192" + "scala:S1192" // Noncompliant
+  }
+
+  @SuppressWarnings("scala:S1193") // Compliant
+  object Reproducer {
+    @SuppressWarnings("scala:S1193")
+    def a(): String =  "Hello!" + "Hello!" + "Hello!" // Noncompliant
+    @SuppressWarnings("scala:S1193")
+    def b(): String =  "Welcome!" + "Welcome!" + "Welcome!" // Noncompliant
+  }
```

Comment:
I would add a case where the literal in the annotation AST is nested a little bit more, something like:
```
@Deprecated(since = "some-version") // Compliant
  object Reproducer2 {
    @Deprecated(since = "some-version")
    def a(): String =  ""
    @Deprecated(since = "some-version")
    def b(): String =  ""
  }

  class MyNestedLiteral(values: Array[String]) extends scala.annotation.StaticAnnotation

  @MyNestedLiteral(Array("some-nested-string")) // Compliant
  object Reproducer3 {
    @MyNestedLiteral(Array("some-nested-string"))
    def a(): String =  ""
    @MyNestedLiteral(Array("some-nested-string", "some-nested-string", "some-nested-string"))
    def b(): String =  ""
  }
```